administrator/components/com_joomgallery/helpers/file.php

```diff
@@ -289,70 +289,86 @@ public static function chmod($path, $mode, $is_dir = false)
-   *                                          Resize to width/height ratio or free setting
+   * @param   int     $settings               Resize to 0=width,1=height,2=max(width,height) or 3=crop
    * @param   int     $new_width              Width to resize
-   * @param   int     $thumbheight            Height to resize
+   * @param   int     $new_height             Height to resize
    * @param   int     $method                 1=gd1, 2=gd2, 3=im
-   * @param   int     $dest_qual              $config->jg_thumbquality/jg_picturequality
-   * @param   boolean $max_width              true=resize to maxwidth
-   * @param   int     $cropposition           $config->jg_cropposition
-   * @param   int     $angle                  $angle to rotate the created image anticlockwise
+   * @param   int     $dest_qual              Quality of the resized image ($config->jg_thumbquality/jg_picturequality)
+   * @param   int     $cropposition           Only if $settings=3:
+   *                                          image section to use for cropping
+   * @param   int     $angle                  $angle to rotate the resized image anticlockwise
+   * @param   boolean $metadata               true=preserve metadata in the resized image
+   * @param   boolean $anim                   true=preserve animation if any
    * @return  boolean True on success, false otherwise
    * @since   1.0.0
    */
-  public static function resizeImage(&$debugoutput, $src_file, $dest_file, $useforresizedirection,
-                                     $new_width, $thumbheight, $method, $dest_qual, $max_width = false, $cropposition, $angle = 0)
+  public static function resizeImage(&$debugoutput, $src_file, $dest_file, $settings,
+                                     $new_width, $new_height, $method, $dest_qual, $cropposition = false, $angle = 0, $metadata = false, $anim = false)
   {
+
+    // animated gifs: https://phpimageworkshop.com/tutorial/5/manage-animated-gif-with-imageworkshop.html
+
     $config = JoomConfig::getInstance();
 
-    // Ensure that the pathes are valid and clean
+    // Ensure that the paths are valid and clean
     $src_file  = JPath::clean($src_file);
     $dest_file = JPath::clean($dest_file);
 
-    // Doing resize instead of thumbnail, copy original and remove it.
-    $imginfo = getimagesize($src_file);
-
-    if(!$imginfo)
+    // Analysis of the source image, if image is valid
+    //$imginfo = getimagesize($src_file, $src_metainfo);
+    if(!($imginfo = JoomFile::analyseImage($src_file)))
     {
       $debugoutput .= JText::_('COM_JOOMGALLERY_UPLOAD_FILE_NOT_FOUND').'<br />';
-
       return false;
     }
 
-    // GD can only handle JPG & PNG images
-    if(    $imginfo[2] != IMAGETYPE_JPEG
-       &&  $imginfo[2] != IMAGETYPE_PNG
-       &&  $imginfo[2] != IMAGETYPE_GIF
+    // GD can only handle JPG, PNG and GIF images
+    if(    $imginfo['type'] != 'JPG'
+       &&  $imginfo['type'] != 'PNG'
+       &&  $imginfo['type'] != 'GIF'
```

Comment:
assotiative array "imginfo" is created in function JoomFile::analyseImage and has another structure. Therefore this chacks has to be done different